app/ide-desktop/lib/common/src/utilities/data/newtype.ts

```diff
@@ -1,11 +1,13 @@
 /** @file Emulates `newtype`s in TypeScript. */
 
+/* eslint-disable @typescript-eslint/consistent-type-definitions */
+
 // ===============
 // === Newtype ===
 // ===============
 
 /** An interface specifying the variant of a newtype. */
-export interface NewtypeVariant<TypeName extends string> {
+type NewtypeVariant<TypeName extends string> = {
```

Comment:
I assume you want to forbid the flexibility which interface gives us?

Also, the types are no longer exported; are they actually used?

Reply:
> forbid the flexibility which interface gives us?

i guess technically? but it's not the main reason

> the types are no longer exported; are they actually used?

only internally - this is the real reason behind this change. apparently if we don't export the `interfaces`, TS complains because it cannot name them (because interface declarations are kinda sorta nominal because they can be merged into)

so basically this change is so that we no longer have to export it (and clutter up autocomplete) just so that it's visible